### ISSUE TITLE
Unbounded page width behaves like 0 available columns

### ISSUE THREAD

**Issue by Gabriel439** [quchen/prettyprinter#49]
According to the documentation of `softline`:

> softline behaves like space if the resulting output fits the page, otherwise like line.

... and this behaves correctly if the document is rendered using the `defaultLayoutOptions` (i.e. 80 columns):

```bash
$ nix-shell --packages 'haskellPackages.ghcWithPackages (pkgs: [ pkgs.prettyprinter ])'
...
$ ghc-pkg list | grep prettyprinter
    prettyprinter-1.1.1
$ ghci
...
Prelude> import Data.Text.Prettyprint.Doc
Prelude Data.Text.Prettyprint.Doc> :set -XOverloadedStrings
Prelude Data.Text.Prettyprint.Doc> let x = "a" <> softline <> "b"
Prelude Data.Text.Prettyprint.Doc> layoutSmart defaultLayoutOptions x
SChar 'a' (SChar ' ' (SChar 'b' SEmpty))

```

... but when I render with the `Unbounded` page width then a line is inserted:

```bash
Prelude Data.Text.Prettyprint.Doc> layoutSmart (defaultLayoutOptions { layoutPageWidth = Unbounded }) x
SChar 'a' (SLine 0 (SChar 'b' SEmpty))
```

... which is surprising since I would interpret `Unbounded` to mean that it would never run out of columns and therefore never need to insert a line according to the `softline` documentation.  Also, the documentation for `Unbounded` seems to confirm that `softline` should not be inserting a line break:

> Layouters should not introduce line breaks on their own

**Comment by quchen** [comment]
`layoutPretty` does not have this problem, only `layoutSmart`.

**Comment by quchen** [comment]
Hah, it’s simply a False instead of a True inside layoutSmart. Simple oversight, simple fix.

**Comment by quchen** [comment]
Done! Released as `prettyprinter-1.2` on Hackage. Thanks for the report! Glad to see `layoutSmart` used, it hasn’t gotten much exposure/testing from my end yet. Dhall as a revdep is pretty cool to have!
